connection.go

```diff
@@ -679,6 +681,22 @@ func (s *connection) ConnectionState() ConnectionState {
 	return s.connState
 }
 
+func (s *connection) updateMaxDatagramDataSize() {
+	if s.peerParams == nil {
+		return
+	}
+	if !s.supportsDatagrams() {
+		return
+	}
+
+	maxDatagramFrameSize := min(s.peerParams.MaxDatagramFrameSize, s.packer.MaxPayloadSize(s.mtuDiscoverer.CurrentSize()))
+	maxDatagramDataSize := (&wire.DatagramFrame{DataLenPresent: true}).MaxDataLen(maxDatagramFrameSize, s.version)
+
+	s.connStateMutex.Lock()
+	s.connState.MaxDatagramSize = int(maxDatagramDataSize)
```

Comment:
Have we consider using atomic uint32?

Reply:
I'm not convinced this would be a worthwhile change, right now there are lower hanging fruit to address first, and if connStateMutex contention turns out to be a problem, I think we might want to revisit how all of ConnectionState is accessed, not just this field.